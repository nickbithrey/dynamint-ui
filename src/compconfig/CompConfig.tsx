import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import Selection from './selection';
import Details from './details';

const CompConfig = () => (
    <Switch>
        <Route exact path={'/compconfig'} component={Selection} />
        <Route exact path={'/compconfig/details'} component={Details} />
    </Switch>
);

export default CompConfig;