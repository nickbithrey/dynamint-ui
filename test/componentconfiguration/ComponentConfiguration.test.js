import React from 'react';
import { shallow } from 'enzyme';

import CompConfig from '~/componentconfiguration/ComponentConfiguration';

describe('<ComponentConfiguration /> rendering', () => {

	it('matches snapshot', () => {
		const wrapper = shallow( <CompConfig /> );
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('will render the full set of routes', () => {
		const wrapper = shallow( <CompConfig /> );

		const switchWrapper = wrapper.find('Switch');
		expect(switchWrapper).toHaveLength(1);
		expect(switchWrapper.find('Route')).toHaveLength(2);
	});
		
});