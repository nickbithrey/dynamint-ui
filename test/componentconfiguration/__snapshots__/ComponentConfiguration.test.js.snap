// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentConfiguration /> rendering matches snapshot 1`] = `
<div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/compconfig"
    />
    <Route
      component={[Function]}
      path="/compconfig/:type"
    />
  </Switch>
</div>
`;
